com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/SerializedHDRenderPipelineGlobalSettings.cs

```diff
@@ -0,0 +1,197 @@
+        public SerializedProperty renderPipelineRayTracingResources;
+
+        public SerializedFrameSettings defaultFrameSettings;
+        public SerializedFrameSettings defaultBakedOrCustomReflectionFrameSettings;
+        public SerializedFrameSettings defaultRealtimeReflectionFrameSettings;
+
+        public SerializedProperty volumeProfileDefault;
+        public SerializedProperty volumeProfileLookDev;
+
+        public SerializedProperty lightLayerName0;
+        public SerializedProperty lightLayerName1;
+        public SerializedProperty lightLayerName2;
+        public SerializedProperty lightLayerName3;
+        public SerializedProperty lightLayerName4;
+        public SerializedProperty lightLayerName5;
+        public SerializedProperty lightLayerName6;
+        public SerializedProperty lightLayerName7;
+
+        public SerializedProperty decalLayerName0;
+        public SerializedProperty decalLayerName1;
+        public SerializedProperty decalLayerName2;
+        public SerializedProperty decalLayerName3;
+        public SerializedProperty decalLayerName4;
+        public SerializedProperty decalLayerName5;
+        public SerializedProperty decalLayerName6;
+        public SerializedProperty decalLayerName7;
+
+        public SerializedProperty shaderVariantLogLevel;
+        public SerializedProperty lensAttenuation;
+        public SerializedProperty diffusionProfileSettingsList;
+
+        //RenderPipelineResources not always exist and thus cannot be serialized normally.
+        bool? m_hasEditorResourceHasMultipleDifferentValues;
+        public bool editorResourceHasMultipleDifferentValues
+        {
+            get
+            {
+                if (m_hasEditorResourceHasMultipleDifferentValues.HasValue)
+                    return m_hasEditorResourceHasMultipleDifferentValues.Value;
+
+                if (serializedObject.targetObjects.Length < 2)
+                {
+                    m_hasEditorResourceHasMultipleDifferentValues = false;
+                }
+                else
+                {
+                    m_hasEditorResourceHasMultipleDifferentValues = serializedSettings.Skip(1).Any(t => t.renderPipelineEditorResources != firstEditorResources);
+                }
+
+                return m_hasEditorResourceHasMultipleDifferentValues.Value;
+            }
+        }
+
+        public HDRenderPipelineEditorResources firstEditorResources
+            => serializedSettings[0]?.renderPipelineEditorResources;
+
+        private List<HDRenderPipelineGlobalSettings> serializedSettings = new List<HDRenderPipelineGlobalSettings>();
+
+        public SerializedHDRenderPipelineGlobalSettings(SerializedObject serializedObject)
+        {
+            this.serializedObject = serializedObject;
+
+            // do the cast only once
+            foreach (var currentSetting in serializedObject.targetObjects)
+            {
+                if (currentSetting is HDRenderPipelineGlobalSettings)
```

Comment:
you can avoid the as if you put `currentSetting is HDREnderPipelineGlobalSettings hdrpGlobalSettings`